src/web/src/components/Posts/Post.tsx

```diff
@@ -184,6 +184,16 @@ const formatPublishedDate = (dateString: string) => {
   }).format(date);
 };
 
+const parseHostName = (hostname: string) => {
+  // Check if there is a prefix to the hostname
```

Comment:
I think you can simplify this a bit.  You are trying to deal with all possible cases, but we only have a handful that we care about (medium, dev.to, etc).  You can hard code them in here and check if the hostname ends with those strings:

```js
if(hostname.endsWith('medium.com')) {
  return "is-medium";
}
if(hostname.endsWith('dev.to')) {
  return "is-devto";
}
...
// Maybe add a "generic" version for blogs we don't know about specifically...
return 'is-generic';
```

const knownHosts = ['medium.com', 'dev.to', ...];
return 
```